src/main/java/com/mycodefu/vsssfshss/server/NettyServer.java

```diff
@@ -98,4 +100,19 @@ public void sendMessage(ChannelId id, ByteBuf message) {
             e.printStackTrace();
         }
     }
+
+    public void broadcast(ChannelId excludeChannelId, ByteBuf message) {
+        try {
+            Channel excludedChannel = allChannels.find(excludeChannelId);
+            if (excludedChannel != null) {
+                ByteBuf outboundMessage = message.copy();
+                WebSocketFrame frame = new TextWebSocketFrame(outboundMessage);
```

Comment:
Why should we restrict ourselves to a text frame? What if you want to send a file through the server?

Reply:
I did a little test with a browser, and the text frame appears more simply. I don't think we should _restrict_ ourselves to text, but for the chat server I think it makes sense.
I have in mind a further refactor we could do which would make it easier to add different types of server in the future.